### PR TITLE
Static analysis: various defects

### PR DESCRIPTION
**Pull request by harryz2000**
Potential nullptr use, divide by zero, incorrect count calculation, always false condition, etc

### CODE REVIEW

**Comment by mildsunrise** [comment]
the two diffs under `src/rtp` are huge, could it be because of line endings?

**Comment by harryz2000** [comment]
Yeah. I also found it. Looks like a merge issue. I am fixing it.

**Comment by harryz2000** [comment]
Yes. It turns out the line ending of these files are CRLF (different with others) in the repository. I manually changed these two files to be CRLF and it appears working now. 

We should unify the line endings one day IMO.

File: src/aac/aacconfig.h [hunk 1]
```diff
@@ -74,7 +74,7 @@ class AACSpecificConfig
 	static BYTE GetSampleRateIndex(DWORD rate)
 	{
 		//Search
-		for (DWORD i=0;i<sizeof(rates);i++)
+		for (DWORD i=0;i<sizeof(rates)/sizeof(rates[0]);i++)
```
Comment (murillo128) [inline src/aac/aacconfig.h:77]: should we change `rates` to an std::array better?

Comment (harryz2000) [reply src/aac/aacconfig.h]: I am OK for both. But change to std::array, which seems more modern.

File: src/aac/aacconfig.h [hunk 2]
```diff
@@ -58,7 +58,7 @@
 class AACSpecificConfig
 {
 public:
-	static constexpr DWORD rates[13] = {
+	static constexpr std::array<DWORD, 13> rates = {
```
Comment (murillo128) [inline src/aac/aacconfig.h:61]: I don't think that the `13` is needed

**Review by murillo128 — CHANGES_REQUESTED** [review]

File: src/aac/aacconfig.h (hunk 2 above)
Comment (harryz2000) [reply src/aac/aacconfig.h:61]: The std::array needs two template arguments. Using one doesn't compile.

**Review by murillo128 — APPROVED** [review]

**Pull request merged**